### PR TITLE
website: document configuration files and internal values

### PR DESCRIPTION
**Pull request by mitchellh**
We forgot to write the docs for this amazing feature!

### CODE REVIEW

File: website/content/docs/app-config/internal.mdx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+page_title: Internal Values - Application Configuration
+description: |-
+  Internal values are static or dynamic configuration values that can be referenced by environment variable or files to compose values.
+---
+
+# Internal Values
+
+Internal values are static or [dynamic](/docs/app-config/dynamic)
+configuration values that can be referenced by environment variables
+or [files](/docs/app-config/files) to compose values. An example common
```
Comment (evanphx) [inline website/content/docs/app-config/internal.mdx:12]: internal can reference internal too.

**Review by evanphx — CHANGES_REQUESTED** [review]
Should we clarify that internal can reference internal?

File: website/content/docs/app-config/internal.mdx (hunk 1 above)
Comment (mitchellh) [reply website/content/docs/app-config/internal.mdx:12]: yes definitely! will update that

File: website/content/docs/app-config/internal.mdx [hunk 2]
```diff
@@ -0,0 +1,51 @@
+---
+layout: docs
+page_title: Internal Values - Application Configuration
+description: |-
+  Internal values are static or dynamic configuration values that can be referenced by environment variable or files to compose values.
```
Comment (izaaklauer) [inline website/content/docs/app-config/internal.mdx:5]: environment variables (plural)

**Review by izaaklauer — COMMENTED** [review]
These are great features, and i'm glad I know about them now!

**Review by briancain — APPROVED** [review]
Other than the two review comments looks good to me 👍🏻

**Pull request merged**